packages/nextjs/mocks/msw/groq/create-dataset.ts

```diff
@@ -0,0 +1,110 @@
+export type Dataset = Array<DatasetItem>;
+
+export type DatasetItem = {
+  _type?: string;
+  _id?: string;
+  [rest: string]: unknown;
+};
+
+type AnyObject = Record<string, unknown>;
+
+export type DatasetOptions = {
+  /**
+   * This list overrides what gets treated as a document.
+   * Documents are always referenced using reference types, instead of being inline.
+   */
+  documentTypes?: string[];
+  /**
+   * This list overrides what gets treated as an inline type.
+   */
+  inlineTypes?: string[];
+};
+
+/**
+ * Builds a flattened dataset from any type of deeply-nested data.
+ */
+export function createDataset(data: Array<AnyObject>, options: DatasetOptions = {}) {
```

Comment:
This is probably the highlight of the PR.
This takes any data, and maps it to Sanity's "dataset" structure. 

This is mostly just mapping objects into references, which is done automatically for data with `_id` attributes.